miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/analytics/MiniAppAnalytics.kt

```diff
@@ -17,34 +18,87 @@ private inline fun <T> whenHasAnalytics(callback: () -> T) {
 
 /** Only init when analytics dependency is provided. */
 @Suppress("SwallowedException", "TooGenericExceptionCaught")
-internal class MiniAppAnalytics(val rasProjectId: String) {
+internal class MiniAppAnalytics(
+    val rasProjectId: String,
+    private val configs: List<MiniAppAnalyticsConfig>
+) {
 
     companion object {
-        var instance: MiniAppAnalytics? = null
-
-        fun init(rasProjectId: String) = whenHasAnalytics {
-            instance = MiniAppAnalytics(rasProjectId)
+        @Suppress("LongMethod")
+        internal fun sendAnalyticsDefault(
+            rasProjectId: String,
+            eType: Etype,
+            actype: Actype,
+            miniAppInfo: MiniAppInfo?
+        ) = try {
+            val params = createParams(
+                rasProjectId = rasProjectId,
+                acc = BuildConfig.ANALYTICS_ACC,
+                aid = BuildConfig.ANALYTICS_AID,
+                actype = actype,
+                miniAppInfo = miniAppInfo
+            )
+            whenHasAnalytics {
+                RatTracker.event(eType.value, params).track()
```

Comment:
The same code is used below for tracking the event. Maybe you should extract it to a method to avoid mis-aligned behavior between the tracking methods? For example:

```
private fun trackEvent(eType: Etype, params: Map<String, Any>) = whenHasAnalytics {
    RatTracker.event(eType.value, params).track()
}
```